src/raw-value.ts

```diff
@@ -239,6 +263,78 @@ export function reverseProxyFactory(env: MembraneBroker) {
+            return key in target;
+        }
+        ownKeys(shadowTarget: ReverseShadowTarget): PropertyKey[] {
+            const { target } = this;
+            // todo: protect against leaking symbols
+            return ownKeys(target);
+        }
+        isExtensible(shadowTarget: ReverseShadowTarget): boolean {
+            if (!ReflectIsExtensible(shadowTarget)) {
+                return false;
+            }
+            const { target } = this;
+            if (!ReflectIsExtensible(target)) {
+                lockShadowTarget(shadowTarget, target);
+                return false;
+            }
+            return true;
+        }
+        getOwnPropertyDescriptor(shadowTarget: ReverseShadowTarget, key: PropertyKey): PropertyDescriptor | undefined {
+            const { target } = this;
+            // todo: leaking key as symbol
+            let rawDesc = ReflectGetOwnPropertyDescriptor(shadowTarget, key);
+            if (!isUndefined(rawDesc)) {
+                return rawDesc;
+            }
+            const secDesc = ReflectGetOwnPropertyDescriptor(target, key);
+            if (isUndefined(secDesc)) {
+                return secDesc;
+            }
+            rawDesc = getRawDescriptor(secDesc);
+            if (secDesc.configurable == false) {
+                // updating the descriptor to non-configurable on the shadow
+                ReflectDefineProperty(shadowTarget, key, rawDesc);
+            }
+            return rawDesc;
+        }
+        getPrototypeOf(shadowTarget: ReverseShadowTarget): RawValue {
+            const { target } = this;
+            return env.getRawValue(ReflectGetPrototypeOf(target));
+        }
+        setPrototypeOf(shadowTarget: ReverseShadowTarget, prototype: RawValue): boolean {
+            const { target } = this;
+            return ReflectSetPrototypeOf(target, env.getSecureValue(prototype));
+        }
+        preventExtensions(shadowTarget: ReverseShadowTarget): boolean {
+            if (ReflectIsExtensible(shadowTarget)) {
+                const { target } = this;
+                if (!ReflectPreventExtensions(target)) {
+                    // if the target is a proxy manually created in the sandbox, it might reject
+                    // the preventExtension call, in which case we should not attempt to lock down
+                    // the shadow target.
+                    return false;
+                }
+                lockShadowTarget(shadowTarget, target);
+            }
+            return true;
+        }
+        defineProperty(shadowTarget: ReverseShadowTarget, key: PropertyKey, rawPartialDesc: PropertyDescriptor): boolean {
+            const { target } = this;
+            // todo: key could be a symbol, and leaked
+            if (ReflectDefineProperty(target, key, getSecurePartialDescriptor(rawPartialDesc))) {
+                // intentionally testing against true since it could be undefined as well
+                if (rawPartialDesc.configurable !== true) {
```

Comment:
Are there other considerations here beyond `configurable`?

Reply:
good question. I don't know. I suspect that if the partial descriptor only have `writable: false || true`, `rawPartialDesc.configurable` will be falsy, which means we will be adding the descriptor to the `shadowTarget`, and that doesn't seem right. The other object is to read the descriptor from the `target` after setting it, to check if we need a copy into the shadowTarget. Feel free to propose something here.

Reply:
any thoughts on this @jdalton ?

Reply:
Would it be easier and avoid potential invariants to just define the descriptor on both?